### PR TITLE
Adding support for Debian package pin priority override to ensure switching to PowerDNS's own package repo

### LINKED ISSUE #74: Debian pinning is lower than the default
Hi,

Could the pinning priority for Debian become a parameter so that it can be adjusted to be higher than the priority set as the default?  At the moment I get the following even though the powerdns::version is set to '4.2'

pdns-recursor:
  Installed: 4.1.11-1+deb10u1
  Candidate: 4.1.11-1+deb10u1
  Version table:
     4.2.2-1pdns.buster 600
        500 http://repo.powerdns.com/debian buster-rec-42/main amd64 Packages
 *** 4.1.11-1+deb10u1 990
        990 http://security.debian.org buster/updates/main amd64 Packages
        100 /var/lib/dpkg/status

Thanks 
Geoff

### PR DESCRIPTION
**Pull request by geoffr**
Added test for pin priority application
Added powerdns::repo::pin_priority with the existing default of 600
Tests all pass https://travis-ci.org/github/geoffr/puppet-powerdns

Closes #74

### CODE REVIEW

File: manifests/repo.pp [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # powerdns::repo
-class powerdns::repo inherits powerdns {
+class powerdns::repo (Integer $pin_priority = $powerdns::params::pin_priority) inherits powerdns {
```
Comment (ju5t) [inline manifests/repo.pp:2]: If I'm not mistaken all of our configurations are done in the top scope at the moment. In this case I think it kind of makes sense to have it here.

Just putting this out there, I'm open for your suggestions on this.

**Comment by brigriffin** [comment]
Do I understand correctly that with this PR the default on Debian would be to install the powerdns package from the APT repo.powerdns.com repo because of a default pin priority of 600 instead of the official Debian APT repo?

**Comment by geoffr** [comment]
> Do I understand correctly that with this PR the default on Debian would be to install the powerdns package from the APT repo.powerdns.com repo because of a default pin priority of 600 instead of the official Debian APT repo?

This is won't change normal behaviour for atm, its an override in case you have "APT::Default-Release" set.   It raises the priority of the Debian packages to 990 which obvious is higher than 600 so this will then allow for overriding that.  Some may also raise the priority of other repo's to ensure correct sourcing of other packages which may interfere with PowerDNS

**Comment by brigriffin** [comment]
@geoffr thanks for clarifying this.

**Comment by ju5t** [comment]
@geoffr would you mind documenting this in the README too?

**Comment by ju5t** [comment]
hi @geoffr, is this still on your radar? Would you mind adding some documentation to this PR?

**Pull request closed without merging**